pkg/oci/utils.go

```diff
@@ -286,6 +287,35 @@ func (spec *CompatOCISpec) PodID() (string, error) {
 	return "", fmt.Errorf("Could not find pod ID")
 }
 
+func ociResourceUpdateRuntimeConfig(ocispec CompatOCISpec, config *RuntimeConfig) error {
+	if ocispec.Linux == nil {
+		return nil
+	}
+
+	if ocispec.Linux.Resources == nil {
+		return nil
+	}
+
+	if ocispec.Linux.Resources.Memory == nil {
+		return nil
+	}
+
+	if ocispec.Linux.Resources.Memory.Limit == nil {
+		return nil
+	}
+
+	memBytes := *ocispec.Linux.Resources.Memory.Limit
+
+	// round up memory bytes to MB
+	mem := int(math.Ceil(float64(memBytes) / 1024 / 1024))
```

Comment:
I still think rounding up is the technically right thing to do. I'd rather hand a container slightly more than requested, rather than artificially starve it. tbh though, the amount we are likely to drop as a % from not rounding up is tiny - given the smallest memory we can probably support might be 128m or so.
If we don't like the float math, we could do something integer like this to round up:
```
mem = (memBytes + ((1024*1024)-1))) / (1024*1024)
```